### DISCUSSION TITLE
Issue when using '@tailwind base;' in NextJS

### DISCUSSION THREAD

**Discussion by smarbo** [tailwindlabs/tailwindcss#14045] [Help]
I am using NextJS with typescript and tailwind and postcss.
I recently, and quite randomly, stumbled upon a problem, error shown below:
⨯ ./app/globals.css:4:1
Module not found: Can't resolve './${url}'

https://nextjs.org/docs/messages/module-not-found

Import trace for requested module:
./app/globals.css
Here is my globals.css file:
```
@tailwind components;
@tailwind base;
@tailwind utilities;

::selection {
	background: magenta;
	color: #ffffff;
}

@keyframes close {
	from {
	  opacity: 1;
		scale: 1;
		rotate: 0;	
	}
	
	to {
		opacity: 0;
		scale: 0.5;
		rotate: 90deg;
	}
}

@keyframes maxi {
	from {
		scale: 1;	
	}

	to {
		transform: translate(0px, 0px);
		width: 100vw;
		height: 100vh;
	}
}

@keyframes mini {
	0% {
		opacity: 1;
		scale: 1;
	}

	100% {
		opacity: 0;
		scale: 0.1;
	}
}

@keyframes open {
	from {
		opacity: 0;
		scale: 0.9;
	}

	to {
		opacity: 1;
		scale: 1;
	}
}

.closing-animation {
	animation: close 0.3s ease-in-out forwards;
}

.opening {
	animation: open 0.3s ease-in-out forwards;
}

.maximized {
	animation: maxi 0.3s ease-in-out forwards;
}

.minimized {
	animation: mini 0.3s ease-in-out forwards !important;
}

html, body {
	margin: 0;
	padding: 0;
	width: 100vw;
	height: 100vh;
	overflow: hidden;
}

* {
	touch-action: none;
}

```
and my tailwind config:
```
import type { Config } from "tailwindcss";

const config: Config = {
  content: [
    "./app/**/*.tsx",
  ],
  theme: {
    extend: {
			backgroundImage: {
				"wallpaper-default": "url('/wallpaper.jpg')",
			}
    },
  },
  plugins: [],
};
export default config;
```

The error randomly appeared while I was working on a component, and, I tried reverting the changes, but it stuck around. I have tried multiple methods such as removing the node_modules, .next folders and running `npm install`, but that did not help at all. I hope there is a way to fix this.
Note: I also found a similar discussion on this page about the same issue but using angular, but my tailwind config didn't seem to have any problems that the user on there had.

**Comment by wongjn** [comment]
As per [the documentation](https://tailwindcss.com/docs/content-configuration#dynamic-class-names):

> The most important implication of how Tailwind extracts class names is that it will only find classes that exist as _complete unbroken strings_ in your source files.
>
> If you use string interpolation or concatenate partial class names together, Tailwind will not find them and therefore will not generate the corresponding CSS:
>
> **Don’t construct class names dynamically**
>
> ```vue
> <div class="text-{{ error ? 'red' : 'green' }}-600"></div>
> ```
> In the example above, the strings `text-red-600` and `text-green-600` do not exist, so Tailwind will not generate those classes.
> Instead, make sure any class names you’re using exist in full:
>
> **Always use complete class names**
>
> ```vue
> <div class="{{ error ? 'text-red-600' : 'text-green-600' }}"></div>
> ```

So in your case, the error most likely comes from some class name you are trying to construct dynamically like:

```
bg-[url(./${url})]
```

Tailwind CSS does not execute your code – it only scans it as plain text. This means it would see the value `url('./${url}')` **literally**, and thus produce some CSS like:

```css
.bg-\[url\(\.\/\$\{url\}\)\] {
  background-image: url(./${url});
}
```
And since you most likely do not have an image file named **exactly** `./${url}`, the build fails.

Instead, consider using an inline `style` attribute value, like:

```jsx
<div style={{ backgroundImage: `url(./${url})` }}>
```

**Reply by smarbo** [reply]
Thank you so much i was searching for hours and was losing my sanity from this mistake.

**Reply by smarbo** [reply]
The funny thing is the bit which had bg-[${url}] or whatever it was actually inside of a comment aswell

**Reply by boblamceo** [reply]
@smarbo how did you reset it when I deleted the line it still showed me the error

**Reply by smarbo** [reply]
> @smarbo how did you reset it when I deleted the line it still showed me the error

Check other places in your code, if your 100% sure there's no more of it, try restarting the development server
